iidm/iidm-reducer/src/test/java/com/powsybl/iidm/reducer/DefaultNetworkReducerTest.java

```diff
@@ -0,0 +1,288 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.reducer;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import com.powsybl.commons.config.InMemoryPlatformConfig;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.datasource.ReadOnlyDataSource;
+import com.powsybl.commons.datasource.ResourceDataSource;
+import com.powsybl.commons.datasource.ResourceSet;
+import com.powsybl.iidm.network.DanglingLine;
+import com.powsybl.iidm.network.Load;
+import com.powsybl.iidm.network.LoadType;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.test.HvdcTestNetwork;
+import com.powsybl.iidm.network.test.ThreeWindingsTransformerNetworkFactory;
+import com.powsybl.iidm.xml.XMLImporter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Mathieu Bague <mathieu.bague at rte-france.com>
+ */
+public class DefaultNetworkReducerTest {
+
+    private FileSystem fileSystem;
+
+    private PlatformConfig platformConfig;
+
+    private ReadOnlyDataSource dataSource;
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        fileSystem = Jimfs.newFileSystem(Configuration.unix());
+        platformConfig = new InMemoryPlatformConfig(fileSystem);
+
+        dataSource = new ResourceDataSource("eurostag-tutorial1-lf", new ResourceSet("/", "eurostag-tutorial1-lf.xml"));
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        fileSystem.close();
+    }
+
+    @Test
+    public void testLoad() {
+        Network network = new XMLImporter(platformConfig).importData(dataSource, new Properties());
+
+        NetworkReducerObserverImpl observer = new NetworkReducerObserverImpl();
+
+        NetworkReducer reducer = new DefaultNetworkReducerBuilder()
+                .setNetworkFilter(new IdentifierNetworkFilter(Collections.singleton("P1")))
+                .addObserver(observer)
+                .build();
```

Comment:
Maybe we can add default impl builder to interface to have something like this and also having a of method  on IdentifierNetworkFilter.
NetworkReducer reducer = NetworkReducer.builder()
                .withNetworkFilter(IdentifierNetworkFilter.of("P1"))
                .withObserver(observer)
                .build();